supporting_scripts/analysis-of-endpoint-connections/src/main/java/de/tum/cit/endpointanalysis/EndpointAnalyzer.java

```diff
@@ -0,0 +1,125 @@
+
+    private static String EndpointAnalysisResultPath = "endpointAnalysisResult.json";
+
+    private static final Logger logger = LoggerFactory.getLogger(EndpointAnalyzer.class);
+
+    public static void main(String[] args) {
+        analyzeEndpoints();
+        printEndpointAnalysisResult();
+    }
+
+    /**
+     * Analyzes endpoints and matches them with REST calls.
+     *
+     * This method reads endpoint and REST call information from JSON files,
+     * compares them to find matching REST calls for each endpoint, and writes
+     * the analysis result to a JSON file. Endpoints without matching REST calls
+     * are also recorded.
+     */
+    private static void analyzeEndpoints() {
+        ObjectMapper mapper = new ObjectMapper();
+
+        try {
+            List<EndpointClassInformation> endpointClasses = mapper.readValue(new File(EndpointParser.ENDPOINT_PARSING_RESULT_PATH),
+                    new TypeReference<List<EndpointClassInformation>>() {
+                    });
+            List<RestCallFileInformation> restCallFiles = mapper.readValue(new File(EndpointParser.REST_CALL_PARSING_RESULT_PATH),
+                    new TypeReference<List<RestCallFileInformation>>() {
+                    });
+
+            List<UsedEndpoints> endpointsAndMatchingRestCalls = new ArrayList<>();
+            List<EndpointInformation> unusedEndpoints = new ArrayList<>();
+
+            Map<String, List<RestCallInformation>> restCallMap = new HashMap<>();
+
+            // Populate the map with rest calls
+            for (RestCallFileInformation restCallFile : restCallFiles) {
+                for (RestCallInformation restCall : restCallFile.restCalls()) {
+                    String restCallURI = restCall.buildComparableRestCallUri();
+                    restCallMap.computeIfAbsent(restCallURI, k -> new ArrayList<>()).add(restCall);
+                }
+            }
+
+            for (EndpointClassInformation endpointClass : endpointClasses) {
+                for (EndpointInformation endpoint : endpointClass.endpoints()) {
+
+                    String endpointURI = endpoint.buildComparableEndpointUri();
+                    List<RestCallInformation> matchingRestCalls = restCallMap.getOrDefault(endpointURI, new ArrayList<>());
+
+                    // Check for wildcard endpoints if no exact match is found
+                    if (matchingRestCalls.isEmpty() && endpointURI.endsWith("*")) {
+                        for (String uri : restCallMap.keySet()) {
+                            if (uri.startsWith(endpoint.buildComparableEndpointUri().substring(0, endpoint.buildComparableEndpointUri().length() - 1))
+                                    && endpoint.getHttpMethod().toLowerCase().equals(restCallMap.get(uri).get(0).getMethod().toLowerCase())) {
+                                matchingRestCalls.addAll(restCallMap.get(uri));
+                            }
+                        }
+                    }
```

Comment:
This could be a method with the name of the comment

Reply:
I think it would be better, but you do not necessarily need to change it for my approve